packages/core/src/tracing/instrumentation/test/squareCalc.js

```diff
@@ -0,0 +1,60 @@
+/*
+ * (c) Copyright IBM Corp. 2024
+ */
+
+'use strict';
+
+const { EXIT } = require('../../constants');
+const tracingUtil = require('../../tracingUtil');
+const cls = require('../../cls');
+const hook = require('../../../util/hook');
+// eslint-disable-next-line no-unused-vars
+let active = false;
+
+exports.activate = function activate() {
+  active = true;
+};
+
+exports.deactivate = function deactivate() {
+  active = false;
+};
+
+exports.init = function init() {
+  hook.onModuleLoad('square-calc', instrument);
+};
+
+/**
+ * Instruments the 'calculateSquare' method of the 'square-calc' module.
+ * @param {{ calculateSquare: (...args: any[]) => undefined; }} orgModule - The original module.
+ */
+function instrument(orgModule) {
+  const originalCalculateSquare = orgModule;
+
+  orgModule = function () {
```

Comment:
It would have been better to export a set of functions from the esm example library, because you cannot use shimmer.